### PR TITLE
User/morgilad/extend trigger completion list

### PR DESCRIPTION
**Pull request by morgilad**
trigger completion when (_._ and _(_) keyboard keys are pressed.

[link to bug](https://msazure.visualstudio.com/One/_workitems/edit/26299925)

Also improved local environment:
- format initial editor value
- support changes to editor value in dev env by taking the editor value from local storage

### CODE REVIEW

File: package/test/test.js [hunk 1]
```diff
@@ -34,6 +38,10 @@ fetch('./test/mode.txt')
                 },
             });
 
+            editor.onDidChangeModelContent(() => {
```
Comment (sagivf) [inline package/test/test.js:41]: This is just for testing so not a big deal, but is this debounced?

**Review by sagivf — DISMISSED** [review]

File: package/test/test.js [hunk 2]
```diff
@@ -23,8 +23,12 @@ fetch('./test/mode.txt')
         require.config(dev ? loader_dev_config : loader_release_config);
         // 'vs/editor/editor.main' is not required here, but makes things load a little faster
         require(['vs/editor/editor.main', 'vs/language/kusto/monaco.contribution'], function () {
+            const defaultValue = 'StormEvents \n| project StartTime , State \n| where State contains "Texas" \n| count';
+            const storageValue = localStorage.getItem('dev-kusto-query');
+            const value = storageValue?.trim().length ? storageValue : defaultValue;
```
[the remark this answers is not in the document]
Comment (morgilad) [reply package/test/test.js:28]: yes

File: package/test/test.js (hunk 1 above)
Comment (morgilad) [reply package/test/test.js:41]: You are totally correct.
I will add debounce for that.

File: package/test/test.js [hunk 3]
```diff
@@ -34,6 +38,20 @@ fetch('./test/mode.txt')
                 },
             });
 
+            const updateEditorValueInLocalStorage = () => {
```
Comment (sagivf) [inline package/test/test.js:41]: might as well put this in the IIFE as well

Comment (sagivf) [reply package/test/test.js:41]: I think we should use lodash, but if it's an issue I would keep this in some util function.

**Review by sagivf — APPROVED** [review]

**Pull request merged**